framework/src/media/media_player.c

```diff
@@ -0,0 +1,320 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <media/media_player.h>
+#include <pthread.h>
+
+#include <tinyara/audio/pcm.h> // For Testing
+
+#define PLAY_LOCK() do { pthread_mutex_lock(&g_pc.mutex_play); } while(false)
+#define PLAY_UNLOCK() do { pthread_mutex_unlock(&g_pc.mutex_play); } while(false)
+
+#define MAX_HEAD_SIZE 1000
+struct media_play_context_s {
+	pthread_t           pth;
+	bool                thread_running;
+	pthread_mutexattr_t mutex_attr_play;
+	pthread_mutex_t     mutex_play;
+	sem_t               sem_pause;
+	struct pcm         *pcmout;
+	unsigned int        buffer_size;
+	int                 fd;
+	int		    download_size;
+	media_format_e      format;
+	play_state_t        state;
+};
+
+static struct media_play_context_s g_pc;
+int threadfunc_media_playing(void *args);

```

Comment:
I think media_playing is enough. threadfunc is redundant.

Reply:
Fixed to media_playing.